aptos-move/framework/src/natives/hash.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use move_deps::{
+    move_binary_format::errors::PartialVMResult,
+    move_core_types::{
+        gas_schedule::{GasAlgebra, GasCost},
+        vm_status::sub_status::NFE_BCS_SERIALIZATION_FAILURE,
+    },
+    move_vm_runtime::native_functions::NativeContext,
+    move_vm_types::{
+        gas_schedule::NativeCostIndex,
+        loaded_data::runtime_types::Type,
+        natives::function::{native_gas, NativeResult},
+        pop_arg,
+        values::{values_impl::Reference, Value},
+    },
+};
+use smallvec::smallvec;
+use std::{collections::VecDeque, hash::Hasher};
+
+/// Serialize the MoveValue with BCS and then feed the bytes into SipHasher. This is not
+/// cryptographically secure.
+pub fn native_sip_hash(
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.len() == 1);
+    debug_assert!(args.len() == 1);
```

Comment:
why not make these proper errors? probably would be useful to generalize this across all native functions.

Reply:
this is not error iiuc, it's invariant. something is really broken if it's violated